AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/shim.py

```diff
@@ -0,0 +1,211 @@
+)
+from com_amazonaws_kms.smithygenerated.com_amazonaws_kms.shim import (
+    _sdk_error_to_dafny_error as com_amazonaws_kms_sdk_error_to_dafny_error,
+)
+import module_
+import software_amazon_cryptography_keystore_internaldafny_types
+from software_amazon_cryptography_keystore_internaldafny_types import (
+    CreateKeyInput_CreateKeyInput as DafnyCreateKeyInput,
+    CreateKeyOutput_CreateKeyOutput as DafnyCreateKeyOutput,
+    CreateKeyStoreInput_CreateKeyStoreInput as DafnyCreateKeyStoreInput,
+    CreateKeyStoreOutput_CreateKeyStoreOutput as DafnyCreateKeyStoreOutput,
+    GetActiveBranchKeyInput_GetActiveBranchKeyInput as DafnyGetActiveBranchKeyInput,
+    GetActiveBranchKeyOutput_GetActiveBranchKeyOutput as DafnyGetActiveBranchKeyOutput,
+    GetBeaconKeyInput_GetBeaconKeyInput as DafnyGetBeaconKeyInput,
+    GetBeaconKeyOutput_GetBeaconKeyOutput as DafnyGetBeaconKeyOutput,
+    GetBranchKeyVersionInput_GetBranchKeyVersionInput as DafnyGetBranchKeyVersionInput,
+    GetBranchKeyVersionOutput_GetBranchKeyVersionOutput as DafnyGetBranchKeyVersionOutput,
+    GetKeyStoreInfoOutput_GetKeyStoreInfoOutput as DafnyGetKeyStoreInfoOutput,
+    VersionKeyInput_VersionKeyInput as DafnyVersionKeyInput,
+    VersionKeyOutput_VersionKeyOutput as DafnyVersionKeyOutput,
+)
+from typing import Any
+
+from .errors import (
+    CollectionOfErrors,
+    ComAmazonawsDynamodb,
+    ComAmazonawsKms,
+    OpaqueError,
+    ServiceError,
+)
+
+
+import Wrappers
+import software_amazon_cryptography_keystore_internaldafny_types
+import aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_keystore.client as client_impl
+
+
+def _smithy_error_to_dafny_error(e: ServiceError):
+    """
+    Converts the provided native Smithy-modeled error
+    into the corresponding Dafny error.
+    """
+    if isinstance(
+        e,
+        aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_keystore.errors.KeyStoreException,
+    ):
+        return software_amazon_cryptography_keystore_internaldafny_types.Error_KeyStoreException(
+            message=e.message
+        )
+
+    if isinstance(e, ComAmazonawsDynamodb):
+        return software_amazon_cryptography_keystore_internaldafny_types.Error_ComAmazonawsDynamodb(
+            com_amazonaws_dynamodb_sdk_error_to_dafny_error(e.message)
+        )
+
+    if isinstance(e, ComAmazonawsKms):
+        return software_amazon_cryptography_keystore_internaldafny_types.Error_ComAmazonawsKms(
+            com_amazonaws_kms_sdk_error_to_dafny_error(e.message)
+        )
+
+    if isinstance(e, CollectionOfErrors):
+        return software_amazon_cryptography_keystore_internaldafny_types.Error_CollectionOfErrors(
+            message=e.message, list=e.list
+        )
+
+    if isinstance(e, OpaqueError):
+        return software_amazon_cryptography_keystore_internaldafny_types.Error_Opaque(
+            obj=e.obj
+        )
+
+
+class KeyStoreShim(
```

Comment:
I thought we were only generating shims for Tests. Do we need this in implementation?

Reply:
You're right -- these should only be in tests.
I included these to demonstrate the codegen for them. I had a note in my tracking doc to remove them before launch.
However, I just realized the implementation depends on the `_smithy_error_to_dafny_error` in this file.
It should be easy to move this. I've made a note to refactor this in my tracking doc.